segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/ExtendedS3IntegrationTest.java

```diff
@@ -0,0 +1,127 @@
+import io.pravega.segmentstore.storage.impl.rocksdb.RocksDBConfig;
+import java.io.File;
+import java.net.URI;
+import java.nio.file.Files;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import org.junit.After;
+import org.junit.Before;
+
+/**
+ * End-to-end tests for SegmentStore, with integrated Extended S3 Storage and DurableDataLog.
+ */
+public class ExtendedS3IntegrationTest extends StreamSegmentStoreTestBase {
+    //region Test Configuration and Setup
+
+    private static final int BOOKIE_COUNT = 3;
+    private BookKeeperRunner bookkeeper = null;
+    private String baseDir;
+    private ConcurrentMap<String, AclSize> aclMap = new ConcurrentHashMap<>();
+
+    /**
+     * Starts BookKeeper.
+     */
+    @Before
+    public void setUp() throws Exception {
+        bookkeeper = new BookKeeperRunner(this.configBuilder, BOOKIE_COUNT);
+        bookkeeper.initialize();
+
+        String endpoint = "http://127.0.0.1:9020";
+        URI uri = URI.create(endpoint);
+        Properties properties = new Properties();
+        properties.setProperty("s3proxy.authorization", "none");
+        properties.setProperty("s3proxy.endpoint", endpoint);
+        properties.setProperty("jclouds.provider", "filesystem");
+        properties.setProperty("jclouds.filesystem.basedir", "/tmp/s3proxy");
+        baseDir = Files.createTempDirectory("extendeds3_wrapper").toString();
+
+        this.configBuilder.include(ExtendedS3StorageConfig.builder()
+                                                          .with(ExtendedS3StorageConfig.BUCKET, "kanpravegatest")
+                                                          .with(ExtendedS3StorageConfig.ACCESS_KEY_ID, "x")
+                                                          .with(ExtendedS3StorageConfig.SECRET_KEY, "x")
+                                                          .with(ExtendedS3StorageConfig.URI, "http://localhost:9020"));
+    }
+
+    /**
+     * Shuts down BookKeeper and cleans up file system directory
+     */
+    @After
+    public void tearDown() throws Exception {
+        bookkeeper.close();
+        FileHelpers.deleteFileOrDirectory(new File(baseDir));
+    }
+
+    //endregion
+
+    //region StreamSegmentStoreTestBase Implementation
+
+    @Override
+    protected ServiceBuilder createBuilder(ServiceBuilderConfig builderConfig) {
+        return ServiceBuilder
+                .newInMemoryBuilder(builderConfig)
+                .withCacheFactory(setup -> new RocksDBCacheFactory(builderConfig.getConfig(RocksDBConfig::builder)))
+                .withStorageFactory(setup -> {
+                    StorageFactory f = new ExtendedS3StorageFactory(
+                            setup.getConfig(ExtendedS3StorageConfig::builder), setup.getExecutor());
+                    return new ExtendedS3ListenableStorageFactory(f, builderConfig.getConfig(ExtendedS3StorageConfig::builder), this.aclMap);
```

Comment:
The common code is moved to `StreamSegmentStorageTestBase` which expects this to be `ListenableStorageFactory`.